nailgun/entities.py

```diff
@@ -1000,6 +1003,22 @@ def build_pxe_default(self, synchronous=True, **kwargs):
         response = client.get(self.path('build_pxe_default'), **kwargs)
         return _handle_response(response, self._server_config, synchronous)
 
+    def clone(self, synchronous=True, **kwargs):
+        """Helper to clone an existing provision template
+
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all JSON decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.post(self.path('clone'), **kwargs)
+        return _handle_response(response, self._server_config, synchronous)
```

Comment:
I'd propose to do it like that:
```python
        kwargs = kwargs.copy()  # shadow the passed-in kwargs
        kwargs.update(self._server_config.get_client_kwargs())
        response = client.post(self.path('clone'), **kwargs)
        result = _handle_response(response, self._server_config, synchronous)
        if 'id' not in result:
            raise APIResponseError(
                'Server have not returned cloned provision template ID')
        return ConfigTemplate(id=result['id']).read()
```
In this case you'll receive not just some dictionary, but nailgun entity, like we usually receive for any basic operation like create/update.
So your example will become even simplier:
```python
>>> template = entities.ConfigTemplate().create(True)
>>> new_template = template.clone(data={'name': 'test111'})
>>> new_template
nailgun.entities.ConfigTemplate(nailgun.config.ServerConfig(url=u'https://example.com', verify=False, auth=(u'admin', u'password')), snippet=True, locked=False, template_kind=None, operatingsystem=[], template_combinations=[], location=[], template=u'\U000266d3\u4096\U00029041\U00024c9c\u06bb\U00029f07\u68f8\ub567\U0002176c\u3ccb\u6676\uc0c5\U00027a8f\U00021bf6\U000270d7\U000233ff\U00028075\ud1dc\U0001312d\u3c06\uc58e\ub1bc\u9c0c\ucfe0', organization=[], audit_comment=None, id=400, name=u'test111')
```
But as it's a helper and not basic operation, and we already have `ContentView.copy()` which is not following this approach (by accident or on purpose?) this suggestion is debatable.
@Ichimonji10 , your vision, please?